pilot/pkg/model/context.go

```diff
@@ -207,15 +213,26 @@ func ParseServiceNode(s string) (Proxy, error) {
 	default:
 		return out, fmt.Errorf("invalid node type (valid types: ingress, sidecar, router in the service node %q", s)
 	}
-	out.IPAddress = parts[1]
 
-	// Does query from ingress or router have to carry valid IP address?
-	if net.ParseIP(out.IPAddress) == nil && out.Type == Sidecar {
-		return out, fmt.Errorf("invalid IP address %q in the service node %q", out.IPAddress, s)
+	// Get IP from node id, it's only for backward-compatible, IP should come from metadata
+	if len(parts[1]) > 0 {
+		out.IPAddresses = append(out.IPAddresses, parts[1])
 	}
 
 	out.ID = parts[2]
 	out.Domain = parts[3]
+
+	// Get all IP Addresses from Metadata
+	if ipstr, found := metadata["ISTIO_META_INSTANCE_IPS"]; found {
+		out.IPAddresses, _ = parseIPAddresses(ipstr)
```

Comment:
If `ISTIO_META_INSTANCE_IPS` set wrong format, should we ignore it? So the upper ip from node id can still be used.